persistence/pre_persistence/persistence.go

```diff
@@ -266,9 +279,21 @@ func (m *PrePersistenceContext) GetBlockHash(height int64) ([]byte, error) {
 	return []byte(block.BlockHeader.Hash), nil
 }
 
-func (m *PrePersistenceContext) TransactionExists(transactionHash string) bool {
+func (m *PrePersistenceContext) TransactionExists(transactionHash string) (bool, error) {
 	db := m.Store()
-	return db.Contains(append(TransactionKeyPrefix, []byte(transactionHash)...))
+	return db.Contains(append(TransactionKeyPrefix, []byte(transactionHash)...)), nil
+}
+
+func (m *PrePersistenceContext) StoreTransaction(transactionProtoBytes []byte) error {
```

Comment:
So you combined the blockstore with the transaction indexer.

I haven't seen this done in this way before.

I think you need to be explicit about this in the issue

Reply:
Aka expand the scope or we're gonna see double work

Reply:
`Aka expand the scope or we're gonna see double work`

My personal opinion is that it's always better to reduce the scope and simply do the work in multiple PRs if possible.

I removed it from this commit and moved those changes over to `issues/120/blockstore/txindexer`.

Reply:
Okay sounds good